BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2642,6 +2705,14 @@ public override string GetInfo()
             output.AppendLine($"Min Range: {minStaticLaunchRange} m");
             output.AppendLine($"Max Range: {maxStaticLaunchRange} m");
 
+            missileType = missileType.ToLower();
+
+            if (missileType == "missile" && BDArmorySettings.MISSILE_FUEL_DEDUCTION)
```

Comment:
Use `if (BDArmorySettings.MISSILE_FUEL_DEDUCTION && missileType == "missile")` instead to avoid the string comparison check when it's not necessary.

Reply:
I'd recommend using ``weaponClass == WeaponClasses.Missile`` (or alternatively ``weaponClass != WeaponClasses.Bomb`` if you're intending Torpedoes to also have fuel burn). in place of the ``missileType == "missile"`` string comparator.